include/swift/AST/AccessScope.h

```diff
@@ -52,12 +47,15 @@ class AccessScope {
   /// Returns true if this is a child scope of the specified other access scope.
   bool isChildOf(AccessScope AS) const {
     if (!isPublic() && !AS.isPublic())
-      return allowsAccess(getDeclContext(), AS.getDeclContext());
+      return getDeclContext()->isAccessibleChildOf(AS.getDeclContext());
```

Comment:
This was a helper method because DeclContext really shouldn't know about access. Can you put it back?